### PR TITLE
Feature/598 make java8 work using updated elasticsearch

### LINKED ISSUE #598: Java 8 migration / cleanup
It seams that their is an issue with incompatible guava versions (fulltext search feature)
+--- com.google.guava:guava-base:r03
|    \--- com.google.guava:guava-annotations:r03
+--- org.elasticsearch:elasticsearch:2.3.1
|    +--- com.google.guava:guava:18.0
(List is probably incomplete, please comment if you find another one, thanks!)

Upgrade all 3rd party libs to Java 8 compatible ones:
- RESTEasy
- Javaassist
- guava
- Elasticsearch

This also fixes #616 then.

**Acceptance criteria**
* [ ] Runs as before --> test
* [ ] Runs in docker ---> test?
* [ ] Runs at least with our tomcat versions used --> update elasticsearch (and if needed tomcat) on CI ?
* [ ] In case new tomcat version is needed mention it in documentation and in changelog
* [ ] It is possible to use lambdas/closures in our REST controllers --> test and cleanup in code
* [ ] ElasticSearch works as before --> test!
* [ ] Changelog mentions breaking change to Elasticsearch version (and if needed other breaking changes)
* [ ] Generic logging of all REST request works as before.

### LINKED ISSUE #616: Scenarioo only supports Elasticsearch 2.x, not 5.x
Tried to run Scenarioo with Elasticsearch 5.5.1, but it did not work. Going back to 2.4.6 worked.

Will be fixed by #598 - ses #598 for current state and remaining work.

### PR DESCRIPTION
**Pull request by bruderol**
This branch includes changes from other open PRs !!!

Fixes #598 
Fixes #616 

What is missing:
* [x] fix resources not resolvable anymore due to Rest Easy update
* [x] fix client to work with changes for labels in JSON format due to RestEasy Update
* [x] merge with changes form other branches and register added resources in new RestEasy app startup code
* [x] there seems to be a problem with branch aliases - one test failing, also the dropdown seems broken for aliases
* [x] make this branch pass on CI by installing Elasticsearch 5 on CI
* [x] make clustername configurable or at least choose one less generic than "elasticsearch" - maybe simply "scenarioo" as default ??
* [x] Document update to elastic search 5 and mention in changelog

### CODE REVIEW

**Comment by bruderol** [comment]
@cgrossde @adiherzog do you agree that we can merge this once the build is green? (elasticsearch and rest easy is working now)

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,11 +37,11 @@ as explained in the [Migration Guide](docs/setup/Migration-Guide.md). This can n
 
 * **Internal format of comparisons between builds (DiffViewer feature) has changed:** After an update to the new version you will see no calculated comparisons on old builds anymore and need to reimport those builds for which you want to calculate and see comparisons again. See Step 3 in [Migration Guide](docs/setup/Migration-Guide.md).
 
+* **Full Text Search needs Elastic Search Version 5:** We upgraded the full text search feature to use ElasticSearch 5. If you want to use that feature your have to configure a compatible ElasticSearch server and a cluster name as described in [Full Text Search Setup Guide](docs/features/full-text-search/setup.md)
```
Comment (cgrossde) [inline CHANGELOG.md]: I would give the exact version. AFAIK different versions of library and server don't play along well.

File: docs/contribute/Development-Environment-Setup.md [hunk 2]
```diff
@@ -22,7 +22,7 @@ corresponding most recent markdown file for that page).
  * Java JDK 1.8
  * Tomcat 7 (recommended), 8 or 9
  * node 8.11.2 (comes with npm 5.6.0)
- * Elasticsearch 2.x (see [install guide](../features/full-text-search/setup.md) for details).
+ * Elasticsearch 5.x (see [Full Text Search Setup Guide](../features/full-text-search/setup.md) for details).
```
Comment (cgrossde) [inline docs/contribute/Development-Environment-Setup.md]: Same here, would give a specific version.

File: docs/contribute/ci-server-setup/Elasticsearch-5.md [hunk 3]
```diff
@@ -0,0 +1,42 @@
+## Install and Run with Docker
+
+Install and run using docker:
+
+```
+docker pull docker.elastic.co/elasticsearch/elasticsearch:5.6.9
+docker run -d -name elasticsearch5 -p 9205:9200 -p 9305:9300 -e cluster.name=elasticsearch -e xpack.ml.enabled=false -e xpack.security.enabled=false docker.elastic.co/elasticsearch/elasticsearch:5.6.9 
+```
+
+Set port values that do not conflict with other existing installation (e.g. Elasticsearch2 whoch is currently running in parallel)
```
Comment (cgrossde) [inline docs/contribute/ci-server-setup/Elasticsearch-5.md]: *which

File: docs/contribute/ci-server-setup/Elasticsearch-5.md [hunk 4]
```diff
@@ -0,0 +1,42 @@
+## Install and Run with Docker
```
Comment (cgrossde) [inline docs/contribute/ci-server-setup/Elasticsearch-5.md]: Great docu :) thanks for writing it 👍

File: scenarioo-server/src/main/java/org/scenarioo/rest/application/ScenariooRestApplication.java [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+@ApplyRequestLogging
+public class ScenariooRestApplication extends Application {
+
+	private static final Logger LOGGER = Logger.getLogger(ScenariooWebApplication.class);
+
+	private Set<Object> singletons = new HashSet<Object>();
+
+	public ScenariooRestApplication() {
+		// find . -name \*Resource.java
+		//does the order play a role in what order URLs get matched??
```
Comment (cgrossde) [inline scenarioo-server/src/main/java/org/scenarioo/rest/application/ScenariooRestApplication.java]: Maybe answer this comment or remove it?

File: scenarioo-server/src/main/java/org/scenarioo/rest/scenario/ScenariosResource.java [hunk 6]
```diff
@@ -36,40 +36,41 @@
...
 	private final AggregatedDocuDataReader aggregatedDataReader = new ScenarioDocuAggregationDao(
 			configurationRepository.getDocumentationDataDirectory());
-	
+
 	private final ScenarioDetailsMapper scenarioDetailsMapper = new ScenarioDetailsMapper();
-	
+
 	@GET
 	@Produces({ "application/xml", "application/json" })
+	@Path("")
```
Comment (cgrossde) [inline scenarioo-server/src/main/java/org/scenarioo/rest/scenario/ScenariosResource.java]: This `@Path` is redundant => remove

File: scenarioo-server/src/main/java/org/scenarioo/rest/application/ScenariooRestApplication.java (hunk 5 above)
Comment (bruderol) [reply scenarioo-server/src/main/java/org/scenarioo/rest/application/ScenariooRestApplication.java]: I removed it

File: scenarioo-server/src/main/java/org/scenarioo/rest/scenario/ScenariosResource.java (hunk 6 above)
Comment (bruderol) [reply scenarioo-server/src/main/java/org/scenarioo/rest/scenario/ScenariosResource.java]: done

File: docs/contribute/ci-server-setup/Elasticsearch-5.md (hunk 3 above)
Comment (bruderol) [reply docs/contribute/ci-server-setup/Elasticsearch-5.md]: done

File: docs/contribute/Development-Environment-Setup.md (hunk 2 above)
Comment (bruderol) [reply docs/contribute/Development-Environment-Setup.md]: Ok, will add it

File: CHANGELOG.md (hunk 1 above)
Comment (bruderol) [reply CHANGELOG.md]: ok, will add it

**Pull request merged**